Actions/TroubleShooting/TroubleShoot.Secrets.ps1

```diff
@@ -0,0 +1,76 @@
+﻿Param(
+    [Parameter(HelpMessage = "All GitHub Secrets in compressed JSON format", Mandatory = $true)]
+    [PSCustomObject] $gitHubSecrets,
+    [Parameter(HelpMessage = "Display the name (not the value) of secrets available to the repository", Mandatory = $true)]
+    [bool] $displayNameOfSecrets
+)
+
+#
+# Check GitHub Secrets for common mistakes
+# If GitHub secrets are multi-line, then every line in any GitHub secret available to the workflow will be masked individually by GitHub
+# This can cause problems if these values are used as elsewhere - f.ex. if a line contains a { or a } character, then no JSON string can be parsed from one job to another
+# This function checks for multi-line secrets displays warnings if multi-line secrets with lines containing short strings
+#
+
+# minSecretSize determines the minimum length of a secret value before a warning is displayed
+$minSecretSize = 8
```

Comment:
It is 7 now (minimum size is 8)
and yes - the value of 3 or 4 was mostly a tradeoff to avoid warnings in the CI/CD workflow.
I do think that any password or other secret added to GitHub of length 7 or less is dangerous - if we get feedback on this from partners, we can change it to a lower number (which is better than going higher)